staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -139,7 +140,9 @@ func NewDefaultClientConfigLoadingRules() *ClientConfigLoadingRules {
 
 	envVarFiles := os.Getenv(RecommendedConfigPathEnvVar)
 	if len(envVarFiles) != 0 {
-		chain = append(chain, filepath.SplitList(envVarFiles)...)
+		fileList := filepath.SplitList(envVarFiles)
+		// prevent the same path load multiple times
+		chain = append(chain, slice.RemoveDuplicates(fileList)...)
```

Comment:
Is there nothing that can override values of previous configs?

@ncdc I think you were diving into this a lot. Can you take a look?

Reply:
@sttts I'm not exactly sure what you're asking. Could you elaborate please?

Reply:
The PR is about kubeconfigs which are loaded multiple times. The question is about whether the order matters and if yes which of the duplicates we have to keep (the last or the first).

Reply:
Are you thinking of a scenario like this?

- list is A,B,A
- A gets loaded
- B gets loaded
- A gets loaded again and is mutated (overrides)
- If we keep the first A and remove the second A, we'll lose the mutations?

Reply:
Exactly.

Reply:
Ok, now we're on the same page 😄. I honestly don't know the answer - when I was digging into the loading code before, it was to figure out why in-cluster namespace wasn't being honored.

Reply:
@liggitt do you have any familiarity with Stefan's concern above? Is this safe?

Reply:
they do layer, so if `A` and `B` overlap, `A,B,A` is not the same as `A,B`

I *think* if we remove duplicates, we should remove earlier instances in favor of later instances

Reply:
This is a good argument for simplifying the loading code